src/MatrixClient.ts

```diff
@@ -721,6 +721,10 @@ export class MatrixClient extends EventEmitter {
                     await Promise.resolve(this.storage.setSyncToken(token));
                 }
             } catch (e) {
+                // If we've requested to stop syncing, don't bother checking the error.
+                if (this.stopSyncing) {
+                    return;
```

Comment:
Let's post a log message here, like the one for before the try-catch.

It also might be helpful to make the log message slightly different to indicate which return path was taken.
```suggestion
                if (this.stopSyncing) {
                    LogService.info("MatrixClientLite", "Client stop requested - cancelling sync");
                    return;
```

Reply:
Seems reasonable. I don't know if the distinction is worth it, you don't really care about errors on shutdown?